test/utils.js

```diff
@@ -42,6 +42,32 @@ t.same(deepMap(sample, (v) => v), sample)
 t.same(deepMap(sample, (v) => v, '$'), sample)
 t.same(deepMap(sample, (v) => v, '$', new Set()), sample)
 
+t.test('deepMap error', async (t) => {
+  const error = new Error('test')
+  error.custom = 'custom'
+  error.code = '1234'
+  error.statusCode = 500
+  const result = deepMap(error)
+  t.same(result.errorType, 'Error')
+  t.same(result.err.message, 'test')
+  t.same(result.err.custom, undefined)
+  t.same(result.err.code, '1234')
+  t.same(result.err.statusCode, 500)
+})
+
+t.test('deepMap error nested', async (t) => {
+  const error = new Error('test')
+  error.custom = 'custom'
+  error.code = '1234'
+  error.statusCode = 500
+  const result = deepMap({ meow: error })
```

Comment:
Typically when errors are nested, they're nested as `err` or `error`. It would be good to avoid doubly nesting them.

```suggestion
  const result = deepMap({ message: 'failed to find tuna or salmon', err: error })
```